### PR TITLE
353 check key state before use

### PR DESCRIPTION
**Pull request by Donnie-Ice**
Should `ekid` and `akid` key state be checked in every case?

### CODE REVIEW

File: src/core/crypto_aos.c [hunk 1]
```diff
@@ -359,21 +359,31 @@ int32_t Crypto_AOS_ApplySecurity(uint8_t *pTfBuffer)
 
     // Get Key
     crypto_key_t *ekp = NULL;
+    crypto_key_t *akp = NULL;
     ekp               = key_if->get_key(sa_ptr->ekid);
-    if (ekp == NULL)
+    akp               = key_if->get_key(sa_ptr->akid);
+
+    if (ekp->key_state != KEY_ACTIVE || akp->key_state != KEY_ACTIVE)
```
Comment (dccutrig) [inline src/core/crypto_aos.c:366]: Line 366: Should this be an AND? As in, if you get this far and neither key is active, then there's a problem.

Does this introduce a segfault by not first checking if ekp == null, or akp==null before accessing the key_state?

File: src/core/crypto_aos.c [hunk 2]
```diff
@@ -1200,21 +1210,31 @@ int32_t Crypto_AOS_ProcessSecurity(uint8_t *p_ingest, uint16_t len_ingest, uint8
...
```
Comment (dccutrig) [inline src/core/crypto_aos.c:1217]: Same comments as above in 366.

File: src/core/crypto_tc.c [hunk 3]
```diff
@@ -390,12 +391,19 @@ int32_t Crypto_TC_Do_Encrypt_PLAINTEXT(uint8_t sa_service_type, SecurityAssociat
 
         /* Get Key */
         ekp = key_if->get_key(sa_ptr->ekid);
-        if (ekp == NULL)
+        akp = key_if->get_key(sa_ptr->akid);
+        if (ekp == NULL || akp == NULL)
```
Comment (dccutrig) [inline src/core/crypto_tc.c:395]: Believe this needs to be an AND?

Comment (Donnie-Ice) [reply src/core/crypto_tc.c]: Line 395 was meant to check both keys for null returns of get_key. Just thought I would knock them both out in the same line.

Comment (dccutrig) [reply src/core/crypto_tc.c]: On 395 isn't it valid if one of them isn't NULL? Therefore the other could be null

Comment (Donnie-Ice) [reply src/core/crypto_tc.c]: Yes. A key should only be NULL if the ekid or akid are less than 0 or greater than NUM_KEYS, therefore making it invalid. In any case, the ekid AND akid should be within this range. If either are NULL, then it should throw an error. Correct me if I'm wrong though.

**Review by rjbrown2 — APPROVED** [review]

**Review by dccutrig — APPROVED** [review]

**Pull request merged**